addons/sale_loyalty/models/sale_order.py

```diff
@@ -355,23 +357,33 @@ def _get_reward_values_discount(self, reward, coupon, **kwargs):
             }]
         discount_factor = min(1, (max_discount / discountable)) if discountable else 1
         mapped_taxes = {tax: self.fiscal_position_id.map_tax(tax) for tax in discountable_per_tax}
-        reward_dict = {tax: {
-            'name': _(
-                'Discount: %(desc)s%(tax_str)s',
-                desc=reward.description,
-                tax_str=len(discountable_per_tax) and any(t.name for t in mapped_taxes[tax]) and _(' - On product with the following taxes: %(taxes)s', taxes=", ".join(mapped_taxes[tax].mapped('name'))) or '',
-            ),
-            'product_id': reward.discount_line_product_id.id,
-            'price_unit': -(price * discount_factor),
+        product_dict = {
             'product_uom_qty': 1.0,
             'product_uom': reward.discount_line_product_id.uom_id.id,
             'reward_id': reward.id,
             'coupon_id': coupon.id,
             'points_cost': 0,
             'reward_identifier_code': reward_code,
             'sequence': sequence,
-            'tax_id': [(Command.CLEAR, 0, 0)] + [(Command.LINK, tax.id, False) for tax in mapped_taxes[tax]]
-        } for tax, price in discountable_per_tax.items() if price}
+        }
+        if reward.discount_mode == 'per_order':
+            reward_dict = {'tax': {
+                **product_dict,
+                'name': _("Discount: %(discount)s", discount=reward.description),
+                'product_id': reward.discount_line_product_id.id,
+                'price_unit': -max_discount,
+            }}
+        else:
+            reward_dict = {tax: {
+                **product_dict,
+                'name': _(
+                    'Discount: %(discount)s%(tax_str)s', discount=reward.description,
+                    tax_str=len(discountable_per_tax) and any(t.name for t in mapped_taxes[tax]) and _(' - On product with the following taxes: %(taxes)s', taxes=", ".join(mapped_taxes[tax].mapped('name'))) or '',
```

Comment:
Double quotes for translatable strings in Python :pray: 
```suggestion
                    "Discount: %(discount)s%(tax_str)s", discount=reward.description,
                    tax_str=len(discountable_per_tax) and any(t.name for t in mapped_taxes[tax]) and _(" - On product with the following taxes: %(taxes)s", taxes=", ".join(mapped_taxes[tax].mapped('name'))) or "",
```